docs/source/user-guide/guide_file_fetchers.rst

```diff
@@ -0,0 +1,57 @@
+Implementing a custom file fetcher
+----------------------------------
+
+DKAN uses a library called getdkan/file-fetcher. This library allows developers to extend the file transfer functionality for their specialized needs.
+
+This library is used to download a resource, such as a CSV file, so that it can be loaded into the database and presented through the UI and API. This process is called 'localization,' because the source resource is copied to the local file system.
+
+The standard file fetcher processors will probably be adequate for most uses, but there could be other use cases, such as needing to authenticate, or getting a file from S3 instead of HTTP.
+
+In cases such as these, we might want to add our own processor class to extend the file fetcher functionality.
+
+TL;DR:
+======
+
+- A code example can be found in the ``custom_processor_test`` module, which is used to test this functionality.
+- Implement ``FileFetcher\Processor\ProcessorInterface`` as a custom processor for ``FileFetcher``. Copy or subclass ``FileFetcher\Processor\Local`` and/or ``FileFetcher\Processor\Remote`` if it makes things easier.
+- Create a ``FileFetcherFactory`` class which instantiates a ``FileFetcher`` using configuration specifying the new processor in the processors array. Always merge in the configuration supplied to ``getInstance()``, as well.
+- Specify this new ``FileFetcherFactory`` as a service which decorates ``dkan.common.file_fetcher``.
+
+How to:
+=======
+
+To implement a new file processor, a custom file-fetcher processor class should be created, implementing ``FileFetcher\Processor\ProcessorInterface``. This class could subclass ``FileFetcher\Processor\Remote`` or ``FileFetcher\Processor\Local``, or be a new implementation. See the section below, "How processors work" for some more implementation details.
```

Comment:
Again, I think on such a short doc, all these extra things easing people through the docs are just unnecessary.

```suggestion
To implement a new file processor, a custom file-fetcher processor class should be created, implementing ``FileFetcher\Processor\ProcessorInterface``. This class could subclass ``FileFetcher\Processor\Remote`` or ``FileFetcher\Processor\Local``, or be a new implementation.
```